python/pyarrow/tests/test_pandas.py

```diff
@@ -4082,6 +4082,18 @@ def test_array_to_pandas():
         # tm.assert_series_equal(result, expected)
 
 
+def test_roundtrip_empty_table_with_intervalrange_index():
```

Comment:
```suggestion
def test_roundtrip_empty_table_with_extension_dtype_index():
```

It's not an issue specifically with IntervalDtype, but more specifically with any extension dtype that defines a `__from_arrow__` (and interval type is one of the examples of this in pandas)